server/internal/core/application/covenantless.go

```diff
@@ -752,19 +758,60 @@ func (s *covenantlessService) UpdateTxRequestStatus(_ context.Context, id string
 }
 
 func (s *covenantlessService) SignVtxos(ctx context.Context, forfeitTxs []string) error {
-	return s.forfeitTxs.sign(forfeitTxs)
+	s.currentRoundLock.Lock()
+	defer s.currentRoundLock.Unlock()
+	currentRound := s.currentRound
+
+	if err := s.forfeitTxs.sign(forfeitTxs); err != nil {
+		return err
+	}
+
+	return s.checkForfeitsAndBoardingSigsSent(currentRound)
 }
 
 func (s *covenantlessService) SignRoundTx(ctx context.Context, signedRoundTx string) error {
 	s.currentRoundLock.Lock()
 	defer s.currentRoundLock.Unlock()
+	currentRound := s.currentRound
 
-	combined, err := s.builder.VerifyAndCombinePartialTx(s.currentRound.UnsignedTx, signedRoundTx)
+	combined, err := s.builder.VerifyAndCombinePartialTx(currentRound.UnsignedTx, signedRoundTx)
 	if err != nil {
 		return fmt.Errorf("failed to verify and combine partial tx: %s", err)
 	}
 
 	s.currentRound.UnsignedTx = combined
+
+	return s.checkForfeitsAndBoardingSigsSent(currentRound)
+}
+
+func (s *covenantlessService) checkForfeitsAndBoardingSigsSent(currentRound *domain.Round) error {
+	roundTx, err := psbt.NewFromRawBytes(strings.NewReader(currentRound.UnsignedTx), true)
+	if err != nil {
+		return fmt.Errorf("failed to parse round tx: %w", err)
+	}
+
+	numOfInputsSigned := 0
+	for _, v := range roundTx.Inputs {
+		if len(v.TaprootScriptSpendSig) > 0 {
+			if len(v.TaprootScriptSpendSig[0].Signature) > 0 {
+				numOfInputsSigned++
+			}
+		}
+	}
+
+	// Condition: all forfeit txs are signed and
+	// the number of signed boarding inputs matches
+	// numOfBoardingInputs we expect
+	s.numOfBoardingInputsMtx.RLock()
+	numOfBoardingInputs := s.numOfBoardingInputs
+	s.numOfBoardingInputsMtx.RUnlock()
+	if s.forfeitTxs.allSigned() && numOfBoardingInputs == numOfInputsSigned {
+		select {
+		case s.forfeitsBoardingSigsChan <- struct{}{}:
+		default:
```

Comment:
add a time.Sleep(time.Millisecond) to prevent unexpected behaviors if for some reason wriitng to channel gets stuck

Reply:
i dont get it, can you elaborate how sleeping here can help?

Reply:
```suggestion
		default:
		     time.Sleep(time.Millisecond)
```

Reply:
Done